mempool/mempool.go

```diff
@@ -631,6 +825,99 @@ func (mp *TxPool) FetchTransaction(txHash *chainhash.Hash) (*btcutil.Tx, error)
 	return nil, fmt.Errorf("transaction is not in the pool")
 }
 
+// validateReplacement determines whether a transaction is deemed as a valid
+// replacement of all of its conflicts according to the RBF policy. If it is
+// valid, no error is returned. Otherwise, an error is returned indicating what
+// went wrong.
+//
+// This function MUST be called with the mempool lock held (for reads).
+func (mp *TxPool) validateReplacement(tx *btcutil.Tx,
```

Comment:
Does this not have a unit test? Or is it covered by the `TestRBF` cases?

Reply:
`TestRBF` is the unit test. You can see each test case failing the conditions one by one as you go through them.